packer/all_plugins_storage.go

```diff
@@ -0,0 +1,55 @@
+package packer
+
+import (
+	"fmt"
+
+	pluginsdk "github.com/hashicorp/packer-plugin-sdk/plugin"
+)
+
+type PluginComponentType string
+
+const (
+	PluginComponentBuilder       PluginComponentType = "builder"
+	PluginComponentPostProcessor PluginComponentType = "post-processor"
+	PluginComponentProvisioner   PluginComponentType = "provisioner"
+)
+
+type PluginDetails struct {
+	Name        string
+	Description pluginsdk.SetDescription
+	PluginPath  string
+}
+
+type allPluginsStorage struct {
```

Comment:
Rather than being a structure, I believe this could be a simple type alias, though arguably the type can completely vanish if the global component map is initialised as such. Then we can have two receiver-free functions for insertion/lookup in the map